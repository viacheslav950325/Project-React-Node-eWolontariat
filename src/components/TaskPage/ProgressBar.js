const ProgressBar = (props) => {
    const { completed } = props;
  
    const containerStyles = {
      height: 20,
      width: '90%',
      backgroundColor: "#e0e0de",
      borderRadius: 10,
      margin: 15
    }
  
    const fillerStyles = {
      height: '100%',
      width: `${completed}%`,
      backgroundColor: '#FF7A82',
      borderRadius: 'inherit',
      textAlign: 'right'
    }
  
    const labelStyles = {
      padding: 5,
      color: 'white',
      fontWeight: 'bold'
    }
  
    return (
      <div style={containerStyles}>
        <div style={fillerStyles}>
          <span style={labelStyles}>{`${completed}%`}</span>
        </div>
      </div>
    );
  };
  
  export default ProgressBar;