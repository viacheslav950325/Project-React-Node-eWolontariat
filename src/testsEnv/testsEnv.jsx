import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import { mount, shallow } from "enzyme";
import * as ReactReduxHooks from "react-redux";

export const testingEnvironment = (defaultStoreState, Component) => {
    const defaultStore = configureStore([thunk])(defaultStoreState);
    const useSelectorSpy = jest.spyOn(ReactReduxHooks, "useSelector");
    const useDispatch = jest.spyOn(ReactReduxHooks, "useDispatch");
  
  
    const shallowWrapper = (store = defaultStore, props) => {
      useSelectorSpy.mockImplementation((state) => state(store.getState()));
      useDispatch.mockImplementation(() => store.dispatch);
  
      return shallow(<Component {...props} />);
    };
  
    const mountWrapper = (store = defaultStore, props) => {
      useDispatch.mockRestore();
  
      return mount(
        <Provider store={store}>
          <Component {...props} />
        </Provider>
      );
    };
    return { defaultStore, shallowWrapper, mountWrapper };
  };