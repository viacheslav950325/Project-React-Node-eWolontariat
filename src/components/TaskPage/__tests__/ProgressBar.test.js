import { testingEnvironment } from "../../../testsEnv/testsEnv";
import ProgressBar from "../ProgressBar"

const {shallowWrapper} = testingEnvironment({},ProgressBar);

describe("ProgressBar",()=>{
    it("Should render correctly",()=>{
        const wrapper = shallowWrapper();
        expect(wrapper).toMatchSnapshot();
    });
});