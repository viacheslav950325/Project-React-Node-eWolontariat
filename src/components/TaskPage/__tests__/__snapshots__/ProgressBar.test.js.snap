// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Should render correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#e0e0de",
      "borderRadius": 10,
      "height": 20,
      "margin": 15,
      "width": "90%",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#FF7A82",
        "borderRadius": "inherit",
        "height": "100%",
        "textAlign": "right",
        "width": "undefined%",
      }
    }
  >
    <span
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "padding": 5,
        }
      }
    >
      undefined%
    </span>
  </div>
</div>
`;
